tests/unit/test_k8s_common.py

```diff
@@ -120,3 +121,20 @@ def test_get_secret_password(monkeypatch):
     assert kc.get_secret_password("username") is None
     assert kc.get_secret_password("username") is None
     assert kc.get_secret_password("username") == "secret"
+
+
+@patch("os.listdir")
+@patch("os.remove")
+@patch("os.symlink")
+def test_configure_default_cni(os_symlink, os_remove, os_listdir):
+    os_listdir.return_value = ["05-default.conflist", "10-cni.conflist"]
+    cni = endpoint_from_flag("cni.available")
+    cni.get_config.return_value = {
+        "cidr": "192.168.0.0/24",
+        "cni-conf-file": "10-cni.conflist",
+    }
+    kc.configure_default_cni("test-cni")
+    os_remove.assert_called_once_with("/etc/cni/net.d/05-default.conflist")
+    os_symlink.assert_called_once_with(
+        "10-cni.conflist", "/etc/cni/net.d/05-default.conflist"
+    )
```

Comment:
Removed from [here](https://github.com/charmed-kubernetes/charm-kubernetes-worker/pull/103/files#diff-9e2d1ca5e7af1825b55a7bdc1be7d38b66066959e0f8383683fcf84ee325e99cL19-L38) and added here.